worker/execute/session.py

```diff
@@ -0,0 +1,20 @@
+from util.network import get_local_ip, get_random_port
+
+
+class Session:
+    """
+    Base class for alternative types of job sessions.
+    """
+
+    protocol: str
+    ip: str = get_local_ip()
+    port: int
+
+    def __init__(self, protocol: str = "ws"):
+        self.protocol = protocol
+        self.ip = Session.ip
+        self.port = get_random_port()
```

Comment:
Also it seems strange that `ws` is a passed in argument but `port` and `ip` are not. Should they not all be passed in?

Also are there concerns with `ip` being calculated on one machine and then the `Session` being executed on another? Should the `get_local_ip()` call also be done "just in time"?